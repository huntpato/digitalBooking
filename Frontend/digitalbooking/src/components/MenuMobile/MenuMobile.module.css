.menumobile {
  display: none;
}

.menuhide {
  display: none;
}

@media (max-width: 768px) {
  .menumobile {
    display: none;
  }

  .menuhide {
    display: none;
  }
}

@media (max-width: 480px) {
  .menumobile {
    background-color: var(--background-white);
    display: flex;
    flex-direction: column;
    align-content: center;
    height: 100vh;
    position: fixed;
    left: 40vw;
    width: 60vw;
    z-index: 3;
    -webkit-animation: slide-in-right 0.8s both;
	        animation: slide-in-right 0.8s both;
  }

  .menuhide {
    display: none;
    transition: 0.8;
  }

  .social {
    display: flex;
    flex-direction: row-reverse;
    padding-left: 38vw;
  }

  .socialbar {
    display: flex;
    flex-direction: row;
    gap: 30px;
    padding-bottom: 3vh;
    width: 50vw;
  }

  .hide {
    display: none;
  }


  @-webkit-keyframes slide-in-right {
    0% {
      -webkit-transform: translateX(1000px);
              transform: translateX(1000px);
      opacity: 0;
    }
    100% {
      -webkit-transform: translateX(0);
              transform: translateX(0);
      opacity: 1;
    }
  }
  @keyframes slide-in-right {
    0% {
      -webkit-transform: translateX(1000px);
              transform: translateX(1000px);
      opacity: 0;
    }
    100% {
      -webkit-transform: translateX(0);
              transform: translateX(0);
      opacity: 1;
    }
  }
  
  @-webkit-keyframes slide-in-left {
    0% {
      -webkit-transform: translateX(-1000px);
              transform: translateX(-1000px);
      opacity: 0;
    }
    100% {
      -webkit-transform: translateX(0);
              transform: translateX(0);
      opacity: 1;
    }
  }
  
  @keyframes slide-in-left {
    0% {
      -webkit-transform: translateX(-1000px);
              transform: translateX(-1000px);
      opacity: 0;
    }
    100% {
      -webkit-transform: translateX(0);
              transform: translateX(0);
      opacity: 1;
    }
  }
  
}
