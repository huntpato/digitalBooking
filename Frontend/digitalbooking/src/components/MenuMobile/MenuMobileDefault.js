import React, { useContext, useState, useEffect } from "react";
import styles from "./MenuMobileDefault.module.css";
import { Link } from "react-router-dom";
import { useLocation } from "react-router-dom";
import { UserContext } from "../UserContext/UserContext";
import { MobileVisibilty } from "../Nav/MobileVisibility";

export const MenuMobileDefault = () => {
  const { header, logout, buttons, hide, button } = styles;
  const [login, setlogin] = useState();
  const [register, setregister] = useState();
  const location = useLocation();
  const { visibility, setVisibility } = useContext(MobileVisibilty);
  const { user } = useContext(UserContext);

  useEffect(() => {
    if (location.pathname === "/login") {
      setlogin(true);
      setregister(false);
    } else if (location.pathname === "/register") {
      setlogin(false);
      setregister(true);
    } else {
      setlogin(false);
      setregister(false);
    }
  }, [location, login, register, setlogin, setregister]);

  const handleClick = () => {
    setVisibility(false);
  };

  return (
    <div>
      <section className={header}>
        <button className={logout} onClick={handleClick}>
          X
        </button>
        <h4>MENÚ</h4>
      </section>
      <section className={buttons}>
        <Link to="/register" className={register ? hide : button}>
          <h5>Crear cuenta</h5>
        </Link>
        <Link to="/login" className={login ? hide : button}>
          <h5>Iniciar sesión</h5>
        </Link>
      </section>
    </div>
  );
};
