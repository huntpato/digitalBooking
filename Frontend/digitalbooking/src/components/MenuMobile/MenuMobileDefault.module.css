.menumobile {
    display: none;
  }
  
  .menuhide {
    display: none;
  }
  
  @media (max-width: 768px) {
    .menumobile {
      display: none;
    }
  
    .menuhide {
      display: none;
    }
  }
  
  @media (max-width: 480px) {
    .menumobile {
      background-color: var(--background-white);
      display: flex;
      flex-direction: column;
      align-content: center;
      height: 100vh;
      position: fixed;
      left: 40vw;
      width: 60vw;
      z-index: 3;
      -webkit-animation: slide-in-right 0.8s both;
              animation: slide-in-right 0.8s both;
    }
  
    .menuhide {
      display: none;
      transition: 0.8s;
    }
  
    .header {
      align-items: flex-end;
      background-color: var(--principal-color);
      display: flex;
      flex-direction: row-reverse;
      height: 20%;
    }
  
    .header h4 {
      color: var(--background-white);
      padding: 0 5px 5px 0;
    }
  
    .logout {
      background-color: transparent;
      border: none;
      color: var(--background-white);
      bottom: 12vh;
      right: 51vw;
      position: relative;
    }
  
    .buttons {
      align-self: flex-end;
      display: flex;
      gap: 30px;
      height: 85vh;
      flex-direction: column;
      padding: 25px;
    }
  
    .button {
      color: var(--color-typo);
      text-decoration: none;
    }
  

    .hide {
      display: none;
    }
  
    @-webkit-keyframes slide-in-right {
      0% {
        -webkit-transform: translateX(1000px);
                transform: translateX(1000px);
        opacity: 0;
      }
      100% {
        -webkit-transform: translateX(0);
                transform: translateX(0);
        opacity: 1;
      }
    }
    @keyframes slide-in-right {
      0% {
        -webkit-transform: translateX(1000px);
                transform: translateX(1000px);
        opacity: 0;
      }
      100% {
        -webkit-transform: translateX(0);
                transform: translateX(0);
        opacity: 1;
      }
    }
    
    @-webkit-keyframes slide-in-left {
      0% {
        -webkit-transform: translateX(-1000px);
                transform: translateX(-1000px);
        opacity: 0;
      }
      100% {
        -webkit-transform: translateX(0);
                transform: translateX(0);
        opacity: 1;
      }
    }
    
    @keyframes slide-in-left {
      0% {
        -webkit-transform: translateX(-1000px);
                transform: translateX(-1000px);
        opacity: 0;
      }
      100% {
        -webkit-transform: translateX(0);
                transform: translateX(0);
        opacity: 1;
      }
    }
    
  }
  