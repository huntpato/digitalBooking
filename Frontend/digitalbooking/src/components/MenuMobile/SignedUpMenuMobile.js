import React, { useContext, useState, useEffect } from "react";
import styles from "./SignedUpMenuMobile.module.css";
import { Link } from "react-router-dom";
import { UserContext } from "../UserContext/UserContext";
import { MobileVisibilty } from "../Nav/MobileVisibility";

export const SignedUpMenuMobile = () => {
  const { header, logout, buttons, hide, button, avatar,avatarDiv, buttonLogout} = styles;
  const [login, setlogin] = useState();
  const [register, setregister] = useState();
  const { visibility, setVisibility } = useContext(MobileVisibilty);
  const { setuser, user } = useContext(UserContext);

  let name = JSON.parse(user).name;
  let auth = JSON.parse(user).authorities[0];
  const firstName = name.split(" ").slice(0, -1).join(" ");
  const lastName = name.split(" ").slice(-1).join(" ");
  const initials =
    firstName.charAt(0).toUpperCase() + lastName.charAt(0).toUpperCase();

  const handleClick = () => {
    setVisibility(false);
  };
  
  const handleLogout = (e) => {
    e.preventDefault();
    localStorage.clear();
    setuser(null);
  };

  return (
    <div>
      <section className={header}>
        <button className={logout} onClick={handleClick}>
          X
        </button>
        <div className= {avatarDiv}>
        <h4 className={avatar}>{initials}</h4>
        <h4>Hola, {name}</h4>
        </div>
      </section>
      <section className={buttons}>
        <div>
        {auth !== "ADMIN" ? <Link to="/preferences" className={register ? hide : button}>
          <h5>Reservas</h5>
        </Link> : <Link to="/administration" className={register ? hide : button}>
          <h5>Administración</h5>
        </Link>}
        </div>
        <Link to="/logout" className={login ? hide : buttonLogout}  onClick={handleLogout}>
          <h5>Cerrar sesión</h5>
        </Link>
      </section>
    </div>
  );
};
