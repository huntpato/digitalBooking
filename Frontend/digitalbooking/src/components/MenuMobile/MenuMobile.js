import React, {useContext, useState, useEffect} from "react"
import styles from './MenuMobile.module.css'
import { MobileVisibilty } from "../Nav/MobileVisibility";
import { useLocation } from 'react-router-dom';
import { UserContext } from "../UserContext/UserContext";
import { SignedUpMenuMobile } from './SignedUpMenuMobile';
import  {MenuMobileDefault} from './MenuMobileDefault'

export const MenuMobile = (props) => {
  const { social, socialbar, menumobile, menuhide, header} = styles; 
  const {visibility, setVisibility}= useContext(MobileVisibilty);
  const [login, setlogin] = useState();
  const [register, setregister] = useState();
  const location = useLocation();
  const { user } = useContext(UserContext);
  let component = user ? <SignedUpMenuMobile /> : <MenuMobileDefault />;
  
  useEffect(() => {
    if(location.pathname === '/login'){
      setlogin(true);
      setregister(false);
    }else if(location.pathname === '/register'){
      setlogin(false);
      setregister(true);
    }else{
      setlogin(false);
      setregister(false);
    }
  }, [location, login, register, setlogin, setregister])

  const handleClick = () => {
    setVisibility(false)
  }
 
  return (
    <div className={ visibility ? menumobile: menuhide}>
      {component}
        <div className={social}>
        <section className={header}>
          </section>
          <div className={ socialbar }>
            <i className="fab fa-facebook"></i>
            <i className="fab fa-linkedin-in"></i>
            <i className="fab fa-twitter"></i>
            <i className="fab fa-instagram"></i>
          </div>
        </div>
      </div>
  );
};
