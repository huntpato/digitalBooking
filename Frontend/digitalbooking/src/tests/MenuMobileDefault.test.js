import React from 'react';
import { shallow } from 'enzyme';
import { MenuMobileDefault } from '../components/MenuMobile/MenuMobileDefault'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';
import { Link } from "react-router-dom";

jest.mock("react-router-dom", () => ({
    useLocation: jest.fn().mockReturnValue({
      pathname: "/",
      search: "",
      hash: "",
      state: null,
      key: "5nvxpbdafa",
    }),
  }));

describe('Pruebas en componente <MenuMobile/>', ()=> {

    const wrapper= shallow(<MenuMobileDefault />);

    test('Comprobar que <MenuMobile/> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const MenuMobile = wrapper.find('section').at(0);
        expect(MenuMobile.hasClass('header')).toBe(true);
    })   

    test('Comprobar que se renderiza el boton para cerrar el menu', ()=>{

        const boton = wrapper.find('button').at(0);
        expect(boton.hasClass('logout')).toBe(true);
    })

    test('Comprobar componentes Link', ()=>{

      const link = wrapper.find('.button').at(0);
      expect(link.props().to).toEqual('/register');

      const link2 = wrapper.find('.button').at(1);
      expect(link2.props().to).toEqual('/login');
  })

})
