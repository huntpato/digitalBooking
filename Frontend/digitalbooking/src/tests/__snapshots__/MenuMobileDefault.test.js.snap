// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <MenuMobile/> Comprobar que <MenuMobile/> renderiza correctamente. 1`] = `
<div>
  <section
    className="header"
  >
    <button
      className="logout"
      onClick={[Function]}
    >
      X
    </button>
    <h4>
      MENÚ
    </h4>
  </section>
  <section
    className="buttons"
  >
    <Component
      className="button"
      to="/register"
    >
      <h5>
        Crear cuenta
      </h5>
    </Component>
    <Component
      className="button"
      to="/login"
    >
      <h5>
        Iniciar sesión
      </h5>
    </Component>
  </section>
</div>
`;
